File: frontend/src/types/comment.ts
export interface Comment {
  id: string;
  animeId: number;
  userId: string;
  username: string;
  userAvatar?: string;
  content: string;
  likes: number;
  dislikes: number;
  createdAt: string;
  updatedAt: string;
  replies?: Comment[];
  parentId?: string;
}

export interface CommentCreate {
  animeId: number;
  content: string;
  parentId?: string;
}

export interface CommentUpdate {
  content: string;
}

export interface CommentReaction {
  commentId: string;
  type: 'like' | 'dislike';
}

File: backend/app/schemas/comment.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class CommentBase(BaseModel):
    content: str = Field(..., min_length=1, max_length=2000)
    parent_id: Optional[int] = None


class CommentCreate(CommentBase):
    anime_id: int
    episode_id: Optional[int] = None


class CommentUpdate(BaseModel):
    content: Optional[str] = Field(None, min_length=1, max_length=2000)


class CommentInDB(CommentBase):
    id: int
    user_id: int
    anime_id: int
    episode_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class Comment(CommentInDB):
    user: dict  # Будет содержать информацию о пользователе
    replies: List["Comment"] = []


class CommentList(BaseModel):
    data: List[Comment]
    total: int
    page: int
    limit: int


# Обновляем модель для поддержки рекурсивных ссылок
Comment.model_rebuild()
